mesonbuild/backend/backends.py

```diff
@@ -1116,32 +1144,9 @@ def extract_dll_paths(cls, target: build.BuildTarget) -> T.Set[str]:
                 bindir = dep.get_pkgconfig_variable('bindir', [], default='')
                 if bindir:
                     results.add(bindir)
-
-            for link_arg in dep.link_args:
-                if link_arg.startswith(('-l', '-L')):
-                    link_arg = link_arg[2:]
-                p = Path(link_arg)
-                if not p.is_absolute():
-                    continue
-
-                try:
-                    p = p.resolve(strict=True)
-                except FileNotFoundError:
                     continue
 
-                for _ in p.parent.glob('*.dll'):
-                    # path contains dlls
-                    results.add(str(p.parent))
-                    break
-
-                else:
-                    if p.is_file():
-                        p = p.parent
-                    # Heuristic: replace *last* occurence of '/lib'
-                    binpath = Path('/bin'.join(p.as_posix().rsplit('/lib', maxsplit=1)))
-                    for _ in binpath.glob('*.dll'):
-                        results.add(str(binpath))
-                        break
+            results.update(filter(None, map(cls.search_dll_path, dep.link_args)))  # pylint: disable=bad-builtin
```

Comment:
Should this just be a list comprehension instead?

This looks like 2 iterations of a `T.Iterable` that could be done in one.

Reply:
Huh? Do you mean : `results.update(dll_path for link_args in dep.link_args for dll_path in cls.search_dll_path(link_args) if dll_path is not None)` ? It seems more obscure to me...

Reply:
And since those are all generators, it is really just one chained iteration....

Reply:
```python
for arg in dep.link_args:
  dll_path = cls.search_dll_path(arg)
  if dll_path is not None:
    results.append(dll_path)
```

Probably no good way for list comprehension.

Reply:
Personally, I still prefer `filter` and `map`... This is probably a matter of taste.